Diff/diff.py

```diff
@@ -8,18 +8,24 @@
 
 
 def read_file_lines(fname):
-    with open(fname, mode="rt", encoding="utf-8", newline=None) as f:
-        lines = f.readlines()
+    with open(fname, mode="rt", encoding="utf-8") as f:
+        lines = f.read().split('\n')
+
+    # Need to insert back the newline characters between lines
+    if len(lines) > 0:
```

Comment:
Since the "split by \n and readd \n" sequence is repeated on line 83, it would make sense to create a function for this.

Reply:
I'm not sure putting `read().split('\n')` in its own function would actually be beneficial. Line 83 only has the split in common.

Reply:
I think @FichteFoll has a point. Both here and line 83 consist of, roughly:

```python
lines = data.split('\n')
if len(lines) > 0:
    for i in range(len(lines) - 1):
        lines[i] += '\n'
````

Something line this would be more clear:

```python
def split_keepends(text):
    lines = text.split('\n')
    if len(lines) > 0:
        for i in range(len(lines) - 1):
            lines[i] += '\n'
    return lines

with open(fname, mode="rt", encoding="utf-8") as f:
    lines = split_keepends(f.read())

b = self.view.substr(sublime.Region(0, self.view.size()))
b = split_keepends(b)
```

Reply:
Ah right, sorry I clearly didn't quite understand the original suggestion.

Reply:
Adding comments to line ranges hasn't been a feature last month, I think :wink: